### PR TITLE
catch pokemon lured by disk on pokestop

### PR DESCRIPTION
**Pull request by ramarro123**
adding the ability to catch a pokemon lured by disk on fort.

had to refactor EncounterResult in a way that i don't like much but i don't want to break too much the compatibility with the previous version (that's why i convert DiskRes to EncRes)

*implemented the call DISK_ENCOUNTER that have to be used instead of ENCOUNTER

the relative issues are:
#151 #149 #204

now, let's see if this time i have mess with git again ;)

### CODE REVIEW

**Comment by ramarro123** [comment]
as a follow up, it's not that easy to step into a pokestop with an active lured pokemon :)

i have finish my lure so i can't test anymore.

that's the result
catturato EEVEE di tipo: disk

tipo: disk
on my log means that the encounterKind = disk

if someone can do a bit of test to confirm, i appreciate

**Comment by igoticecream** [comment]
Lol, i was going to submit this exact pull request. Yes it works, you can construct a catchablepokemon with fortdata and set spawn point it as getFortId!!!

Check out: https://github.com/AeonLucid/POGOProtos/pull/103

**Comment by ramarro123** [comment]
:)

**Comment by JamesPeters98** [comment]
This has been working perfect on a few lures near me.

File: src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java [hunk 1]
```diff
@@ -99,42 +108,50 @@ public CatchablePokemon(PokemonGo api, WildPokemon proto) {
 		this.expirationTimestampMs = proto.getTimeTillHiddenMs();
 		this.latitude = proto.getLatitude();
 		this.longitude = proto.getLongitude();
+		this.encounterKind = EncounterKind.normal;
 	}
 
 	/**
 	 * Instantiates a new Catchable pokemon.
```
Comment (CaptnBlubber) [inline src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java:115]: Would be clearer if you stated here that this is from a Lured Fort

File: src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java [hunk 2]
```diff
@@ -151,18 +168,38 @@ public EncounterResult encounterPokemon() throws LoginFailedException,
...
+				.setPlayerLongitude(api.getLongitude())
+				.setFortId(getSpawnPointId()).build();
+		ServerRequest serverRequest = new ServerRequest(
+				RequestTypeOuterClass.RequestType.DISK_ENCOUNTER, reqMsg);
+		api.getRequestHandler().sendServerRequests(serverRequest);
+		DiskEncounterResponseOuterClass.DiskEncounterResponse response = null;
+		try {
+			response = DiskEncounterResponseOuterClass.DiskEncounterResponse
+					.parseFrom(serverRequest.getData());
+		} catch (InvalidProtocolBufferException e) {
```
Comment (CaptnBlubber) [inline src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java:189]: Why do you use try catch here when the rest of this method already has the a throws Signature for LoginFailedException and RemoteServerException ?

File: src/main/java/com/pokegoapi/api/map/pokemon/EncounterResult.java [hunk 3]
```diff
@@ -17,38 +17,108 @@
...
+	 * Return the status of the encounter
+	 *
+	 * @return status of results
+	 */
+	public abstract EncounterResponse.Status getStatus();
 
-	public boolean wasSuccessful() {
-		return response != null && getStatus() != null && getStatus().equals(EncounterResponse.Status.ENCOUNTER_SUCCESS);
-	}
+	public static class DiskEncounterResult extends EncounterResult {
```
Comment (CaptnBlubber) [inline src/main/java/com/pokegoapi/api/map/pokemon/EncounterResult.java:37]: Not a big fan of Inner Classes, it makes Class Files large (and therefore harder to read/understand/debug) and Code instantiations of Objects uglier.
Personaly I would Have a Seperate file for Bot DiskEncounterResult.java and EncounterResult.java

File: src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java (hunk 2 above)
Comment (ramarro123) [inline src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java:189]: just for consistency with the rest of API.

also i catch InvalidProtocolBufferException and throw RemoteServerException

take a look to the rest of api, it's done in the same way

File: src/main/java/com/pokegoapi/api/map/pokemon/EncounterResult.java (hunk 3 above)
Comment (ramarro123) [inline src/main/java/com/pokegoapi/api/map/pokemon/EncounterResult.java:37]: not a big fan of inner as well, but in this case the whole class it's 124 line (including the 2 inners), it won't be too hard to navigate or read that.

if there will be need for other EncounterResponse (tm) i will split it in different classes

**Comment by ryangardner** [comment]
Is the only holdup around the style issue regarding the inner classes? 

I started down a very similar path today to try to get the lured pokemon working - just enabling it you can see a _tremendous_ different in how many pokemon are in a densely lured area. 

If there is any way I can help out I'm willing to do so. This is a really important change to get in

**Comment by Grover-c13** [comment]
Iwas holding off til we had the server rate limit fixed. Needs to have conflicts resolved now, so ill just rework it into a new PR

**Comment by Grover-c13** [comment]
Redone in PR #418, ty for contribution @ramarro123

**Pull request closed without merging**